client/src/components/postCard/Post.tsx

```diff
@@ -0,0 +1,116 @@
+import './post.css';
+import PetsIcon from '@mui/icons-material/Pets';
+import Tooltip from '@mui/material/Tooltip';
+import {
+  useEffect, useRef, useState,
+} from 'react';
+import BtnsPost from './BtnsPost';
+import HoverLikes from './HoverLikes';
+import IPost from '../../Interfaces/post/Ipost';
+import UserPostInfo from './UserPostInfo';
+import Comments from './Comments';
+
+const comments = {
+  id: 1,
+  Comments: [{
+    id: 1,
+    UserId: 3,
+    comment: 'Comment1',
+    image: null,
+    User: {
+      name: 'Kakashi',
+      avatar: 'https://media.tenor.com/fR49OunP59UAAAAC/killua-killua-zoldyck.gif',
+      id: 1,
+    },
+  }, {
+    id: 2,
+    UserId: 2,
+    comment: 'Comment2',
+    User: {
+      name: 'Kakashi',
+      avatar: 'https://media.tenor.com/fR49OunP59UAAAAC/killua-killua-zoldyck.gif',
+      id: 1,
+    },
+    image: 'https://i.pinimg.com/originals/8a/81/ec/8a81ecd8fdd266b3221da325875c0ea8.gif',
+  }],
+};
+
+const Post = ({ post }:IPost) => {
+  const [showComments, setShowComments] = useState(false);
+  const useOutsideClick = (callback: Function) => {
+    const ref = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+      const handleClick = (event: any) => {
+        if (ref.current && !ref.current.contains(event.target)) {
+          callback();
+        }
+      };
+
+      document.addEventListener('click', handleClick);
+
+      return () => {
+        document.removeEventListener('click', handleClick);
+      };
+    }, []);
+
+    return ref;
+  };
```

Comment:
it's a great idea to create your custom hook

i think we may use this one in other place

so you can create `hooks` folder and create a file on it for this hook